src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -369,7 +379,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     ============================================================
   -->
   <Target Name="_GetRestoreProjectStyle"
-    DependsOnTargets="_GetProjectJsonPath"
+    DependsOnTargets="_GetProjectJsonPath;CollectPackageReferences"
```

Comment:
A project is restored as PackageReference project if it was one or more PackageReference items. If references are potentially in an external file then CollectPackageReferences needs to be called to find that out.